mergify_engine/json.py

```diff
@@ -0,0 +1,56 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import json
+import enum
+
+from mergify_engine.actions.merge_base import strict_merge_parameter
+
+
+JSON_TYPES = {}
+
+
+def register_types(cls):
+    JSON_TYPES[cls.__name__] = cls
+
+
+class EnumEncoder(json.JSONEncoder):
+    def default(self, v):
+        if isinstance(v, enum.Enum):
+            register_types(type(v))
+            return {
+                "__enum__": True,
```

Comment:
I'd rather use something like:
```suggestion
                "__pytype__": "enum",
```
As we're likely going to extend this to more than just `Enum` in the future.